src/app/security/maturity-model/components/MobileTableOfContents.tsx

```diff
@@ -0,0 +1,56 @@
+'use client'
+
+import { Listbox as HeadlessUIListbox } from '@headlessui/react'
+import { ArrowsDownUp, CaretDown } from '@phosphor-icons/react/dist/ssr'
+
+import { Icon } from '@/components/Icon'
+import { ListboxButton } from '@/components/ListboxButton'
+import { ListboxOption } from '@/components/ListboxOption'
+import { ListboxOptions } from '@/components/ListboxOptions'
+
+import { coreFunctionsData } from '../data/coreFunctionsData'
+import { scrollToSection } from '../utils/scrollToSection'
+import { useUrlHash } from '../utils/useUrlHash'
+
+const options = coreFunctionsData.map(({ slug, title }) => ({
+  id: slug,
+  name: title,
+}))
+
+export function MobileTableOfContents() {
+  const { updateHash, isSectionActive, getHashFromSlug } = useUrlHash()
+
+  const selectedOption = options.find((option) => isSectionActive(option.id))
+
+  return (
+    <nav aria-label="Table of Contents" className="w-full max-w-sm">
+      <HeadlessUIListbox
+        value={selectedOption}
+        onChange={(value) => handleChange(value.id)}
+      >
+        <ListboxButton>
+          <div className="inline-flex w-full items-center gap-2">
+            <Icon component={ArrowsDownUp} />
+            <span className="grow truncate text-left">
+              {selectedOption?.name || 'Select a section'}
+            </span>
+            <Icon component={CaretDown} size={16} weight="bold" />
+          </div>
```

Comment:
It doesn't make a lot of sense to me to have an `ArrowsDownUp` icon here?

![CleanShot 2024-10-22 at 20 17 30@2x](https://github.com/user-attachments/assets/de64b9c7-5792-47ad-a89a-9495790e6bb6)


```suggestion
          <span className="grow truncate text-left">
            {selectedOption?.name || 'Select a section'}
          </span>
          <Icon component={CaretDown} size={16} weight="bold" />
```